libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -626,17 +626,27 @@ void AP_GPS::update_instance(uint8_t instance)
     // detection to run again
     if (!result) {
         if (tnow - timing[instance].last_message_time_ms > GPS_TIMEOUT_MS) {
-            // free the driver before we run the next detection, so we
-            // don't end up with two allocated at any time
-            delete drivers[instance];
-            drivers[instance] = nullptr;
-            memset(&state[instance], 0, sizeof(state[instance]));
-            state[instance].instance = instance;
-            state[instance].status = NO_GPS;
-            state[instance].hdop = GPS_UNKNOWN_DOP;
-            state[instance].vdop = GPS_UNKNOWN_DOP;
-            timing[instance].last_message_time_ms = tnow;
-            timing[instance].delta_time_ms = GPS_TIMEOUT_MS;
+            // do not try to detect again if type is MAV
+            if (_type[instance] == GPS_TYPE_MAV) {
+                state[instance].status = NO_FIX;
+                state[instance].hdop = GPS_UNKNOWN_DOP;
+                state[instance].vdop = GPS_UNKNOWN_DOP;
+                state[instance].have_vertical_velocity = false;
+                state[instance].have_speed_accuracy = false;
+                state[instance].have_horizontal_accuracy = false;
+                state[instance].have_vertical_accuracy = false;
+            } else {
+                // free the driver before we run the next detection, so we
+                // don't end up with two allocated at any time
+                delete drivers[instance];
+                drivers[instance] = nullptr;
+                memset(&state[instance], 0, sizeof(state[instance]));
+                state[instance].instance = instance;
```

Comment:
Unrelated to this PR, but does this line need to be/should be here?

Reply:
This is needed to correct some UAVCAN shenanigans. It really should just go away and be worked around differently, but UAVCAN blindly copies over the state structure so this is needed to allow it to print the detection method on boot correctly. This is strictly in the realm of dirty hack and should be fixed. (Really by removing the `instance` field from state and moving it into the backends state directly.